static/xref/script.js

```diff
@@ -1,57 +1,57 @@
+import { autocomplete } from './autocomplete.js';
+
 class OptionSelector extends HTMLInputElement {
   constructor() {
     super();
-    this._state = {
-      selectedValues: new Set(),
-      options: new Set(),
-      datalist: document.createElement('datalist'),
-    };
   }
 
-  static get sep() {
-    return [',', undefined, 'Enter'];
+  static get observedAttributes() {
+    return ['data-options'];
   }
 
-  /** @returns {Set<string>} */
+  /** @returns {string[]} */
   get values() {
-    return [...this._state.selectedValues];
+    return Array.from(this.selectedValues);
   }
 
-  connectedCallback() {
-    const { datalist, options } = this._state;
-
-    const name = this.getAttribute('name');
-
-    const elSelections = document.querySelector(`.selections[data-for="${this.name}"]`);
-    if (!elSelections) {
-      throw new Error(`[class="selections"][data-for="${name}"] not found`);
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (name === 'data-options') {
+      this.options = [...new Set(newValue.split('|'))];
+      this.fuse = new Fuse(this.options, {});
     }
-    this._state.elSelections = elSelections;
-
-    this.setAttribute('list', name);
+  }
 
-    datalist.setAttribute('id', name);
-    this.dataset.values.split('|').forEach(value => {
-      options.add(value);
-      const option = document.createElement('option');
-      option.value = value;
-      datalist.appendChild(option);
+  connectedCallback() {
+    this.fuse = new Fuse(this.options, {});
+    const limit = parseInt(this.dataset.limit, 10) || 10;
+    autocomplete({
+      input: this,
+      fetch: (text, update) => {
```

Comment:
using fat arrow here makes this ambiguous...

Reply:
`this` refers to current input element. will make it clear with an assignment and no fat arrow..